Source/objects.cpp

```diff
@@ -4350,6 +4351,53 @@ bool Object::IsDisabled() const
 	return IsAnyOf(static_cast<shrine_type>(_oVar1), shrine_type::ShrineFascinating, shrine_type::ShrineOrnate, shrine_type::ShrineSacred);
 }
 
+Object *ObjectAtPosition(Point position, bool considerLargeObjects)
+{
+	if (!InDungeonBounds(position)) {
+		return nullptr;
+	}
+
+	auto objectId = dObject[position.x][position.y];
+
+	if (objectId > 0 || (considerLargeObjects && objectId != 0)) {
+		return &Objects[abs(objectId) - 1];
+	}
+
+	// nothing at this position, return a nullptr
+	return nullptr;
+}
+
+bool IsObjectAtPosition(Point position)
+{
+	return ObjectAtPosition(position) != nullptr;
+}
+
+bool IsItemBlockingObjectAtPosition(Point position)
+{
+	Object *object = ObjectAtPosition(position);
+	if (object != nullptr && object->_oSolidFlag) {
+		// solid object
+		return true;
+	}
+
+	object = ObjectAtPosition(position + Direction::South);
+	if (object != nullptr && object->_oSelFlag != 0) {
+		// open single tile door
+		return true;
+	}
+
+	object = ObjectAtPosition(position + Direction::SouthEast, false);
+	if (object != nullptr) {
+		Object *otherDoor = ObjectAtPosition(position + Direction::SouthWest, false);
+		if (otherDoor != nullptr && object->_oSelFlag != 0 && otherDoor->_oSelFlag != 0) {
+			// open double tile door

```

Comment:
does this exist?

Reply:
This check is actually looking for two objects along a horizontal diagonal (relative to the screen). The player isn't allowed to drop an item on the tile behind an interactive object, and also can't drop on the tile between/behind two interactive objects.